build-support/bin/release.sh

```diff
@@ -13,21 +13,39 @@ function run_local_pants() {
   ${ROOT}/pants "$@"
 }
 
-function local_version() {
-  run_local_pants --version 2>/dev/null
+function compute_version_suffix() {
+  local readonly sha=$1
+  if [[ -z "${SUFFIXED_VERSION}" ]]; then
+    echo ""
+  else
+    # Python requires integer version suffixes.
+    local readonly short_sha=${sha:0:8}
+    local readonly sha_decimal=$((16#${short_sha}))
+    local readonly abs_sha_decimal=${sha_decimal#-}
+    echo "+${abs_sha_decimal}"
```

Comment:
IIUC, `+...` is the prefix used for local version identifiers, and these need not be numbers: https://www.python.org/dev/peps/pep-0440/#local-version-identifiers - the short_sha would suffice. That said, the semantics of a local version imply patching which these releases won't have, they'll be mainline intermediary. I'm not sure those semantics matter though.

Reply:
+1